express-post-json/index.js

```diff
@@ -0,0 +1,27 @@
+const express = require('express');
+const app = express();
+
+let nextId = 1;
+const grades = {};
+const newArray = [];
+app.get('/api/grades', (req, res) => {
+  for (const prop in grades) {
+    newArray.push(grades[prop]);
+  }
+  res.json(newArray);
+})
+
+app.use(express.json());
+
+app.post('/api/grades', (req, res) => {
+  const newGrade = req.body;
+  newGrade.id = nextId;
+  nextId++;
+  newArray.push(newGrade);
```

Comment:
The new grade should not be added to an array it should be added to the `grades` object using the `id` as the key.
```JAVASCRIPT
grades[newGrade.id] = newGrade;
```